### PR TITLE
feat: introduce 'remove' method for 'near_sdk::store::Lazy' collection

### PR DESCRIPTION
**Pull request by ruseinov**
Fixes https://github.com/near/near-sdk-rs/issues/1218

### CODE REVIEW

**Comment by frol** [comment]
@ruseinov Please, use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) for PR naming, so we can simply squash and merge it. Also, please, name PRs in such a ways that the entry would make sense in the [CHANGELOG](https://github.com/near/near-sdk-rs/blob/master/CHANGELOG.md) (include all the context and be specific, e.g. "`near_sdk::store::Lazy` collection stored value" vs "Lazy values")

@race-of-sloths invite

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@ruseinov Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/ruseinov)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: CHANGELOG.md
Comment (frol) [inline CHANGELOG.md:6]: You don't need to add the CHANGELOG manually, just name the PR in such a way that release-plz can just use the commit/PR title as the CHANGELOG entry as is

**Comment by ruseinov** [comment]
> @ruseinov Please, use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) for PR naming, so we can simply squash and merge it. Also, please, name PRs in such a ways that the entry would make sense in the [CHANGELOG](https://github.com/near/near-sdk-rs/blob/master/CHANGELOG.md) (include all the context and be specific, e.g. "`near_sdk::store::Lazy` collection stored value" vs "Lazy values")
> 
> @race-of-sloths invite

Done

**Review by frol — APPROVED** [review]
@race-of-sloths score 3

**Comment by race-of-sloths** [comment]
@frol Thank you for calling!
    
@ruseinov Thank you for the contribution! Join [Race of Sloths](https://race-of-sloths.com) by simply mentioning me in your comment/PRs description and start collecting Sloth Points through contributions to open source projects.

<details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/ruseinov)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Pull request merged**
